app/src/main/java/com/github/se/stepquest/screens/FriendsListPopUp.kt

```diff
@@ -0,0 +1,101 @@
+package com.github.se.stepquest.screens
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import com.github.se.stepquest.R
+
+data class Friend(val name: String, val profilePictureUrl: String, val status: Boolean)
+
+@Composable
+fun FriendsListDialog(onDismiss: () -> Unit, friendsList: List<Friend>) {
+  val blueThemeColor = colorResource(id = R.color.blueTheme)
+  var showAddFriendScreen by remember { mutableStateOf(false) }
+  Dialog(onDismissRequest = { onDismiss() }) {
+    if (showAddFriendScreen) {
+      AddFriendScreen(
+          onDismiss = { showAddFriendScreen = false }, onSecondScreenDismiss = onDismiss)
```

Comment:
Also maybe rename the different onDismiss functions so that what they do is a bit clearer because lines like this get a bit hard to follow imo